client/state/user-settings/actions.js

```diff
@@ -116,22 +137,40 @@ export const removeUnsavedUserSetting = ( settingName ) => ( {
 export function setUserSetting( settingName, value ) {
 	return ( dispatch, getState ) => {
 		const settings = getUserSettings( getState() );
-
-		if ( get( settings, settingName ) === undefined ) {
+		const settingKey = Array.isArray( settingName ) ? settingName : settingName.split( '.' );
+
+		const maybeSetting = get( settings, settingKey );
+
+		if (
+			maybeSetting === undefined &&
+			/* FIXME: excluding these settings is a workaround which allows
```

Comment:
This note was migrated from [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/me/account/main.jsx#L194-L214).